security-profile/security_profile.go

```diff
@@ -1,193 +1,9 @@
-	defaultActError := (p.Oci.Linux.Seccomp.DefaultAction == specs.ActErrno)
-
-	/* For each syscall to block we browse each syscall list of each Seccomp rule */
-	for syscallNameToBlock, syscallArgsToBlock := range syscallsWithArgsToBlock {
-		blocked := false
-		for syscallRuleIndex, syscallRule := range p.Oci.Linux.Seccomp.Syscalls {
-			switch syscallRule.Action {
-			case specs.ActAllow, specs.ActTrace, specs.ActTrap:
-				for syscallNameIndex, syscallName := range syscallRule.Names {
-					/* We found the syscall in the syscall list in a rule and arguments are identical */
-					if syscallName == syscallNameToBlock &&
-						((len(syscallArgsToBlock) == 0 && len(syscallRule.Args) == 0) ||
-							reflect.DeepEqual(syscallRule.Args, syscallArgsToBlock)) {
-
-						/* If this is the only one, just remove that rule from the Seccomp config */
-						if len(p.Oci.Linux.Seccomp.Syscalls[syscallRuleIndex].Names) == 1 {
-							p.Oci.Linux.Seccomp.Syscalls = append(
-								p.Oci.Linux.Seccomp.Syscalls[0:syscallRuleIndex],
-								p.Oci.Linux.Seccomp.Syscalls[syscallRuleIndex+1:]...,
-							)
-
-							break
-						}
-
-						/* Otherwise, remove it from the rule */
-						p.Oci.Linux.Seccomp.Syscalls[syscallRuleIndex].Names = append(
-							p.Oci.Linux.Seccomp.Syscalls[syscallRuleIndex].Names[0:syscallNameIndex],
-							p.Oci.Linux.Seccomp.Syscalls[syscallRuleIndex].Names[syscallNameIndex+1:]...,
-						)
-					}
-				}
-
-			case specs.ActErrno, specs.ActKill:
-				for _, syscallName := range syscallRule.Names {
-					/* We found the syscall in the syscall list in a rule */
-					if syscallName == syscallNameToBlock {
-						blocked = true
-
-						/* We'll keep looking in next rules outside of this loop as
-						 * SECCOMP_RET_TRAP has precedence over SECCOMP_RET_ERRNO for example
-						 * and we want to make sure we remove all those occurrences
-						 */
-						break
-					}
-				}
-			}
-		}
-
-		/* If we don't find it in a blocking rule and default behavior is not to deny, we add one. */
-		if !blocked && !defaultActError {
-			newRule := specs.LinuxSyscall{
-				Names:  []string{syscallNameToBlock},
-				Action: specs.ActErrno,
-				Args:   syscallArgsToBlock,
-			}
-			p.Oci.Linux.Seccomp.Syscalls = append(p.Oci.Linux.Seccomp.Syscalls, newRule)
-		}
-
-	}
-}
-
-/* Block a list of syscalls without specific arguments */
-func (p *Profile) BlockSyscalls(syscallsToBlock ...string) {
-	syscallsWithNoArgsToBlock := make(map[string][]specs.LinuxSeccompArg)
-	for _, syscallsToBlock := range syscallsToBlock {
-		syscallsWithNoArgsToBlock[syscallsToBlock] = []specs.LinuxSeccompArg{}
-	}
-
-	p.BlockSyscallsWithArgs(syscallsWithNoArgsToBlock)
+/* Profile is an abstract interface which represents security profiles.
+ * Each security profile has its own type and its own API as needs
+ * may vary across different profile formats.
+ */
+type Profile interface {
+	GetType() string
```

Comment:
Also - I wonder if there are common APIs we want to pull out into `Profile` for now?  At the very least, `GetType`?